app/graphql/resolvers/submissions_resolver.rb

```diff
@@ -43,11 +26,39 @@ def run
 
   private
 
+  def compute_error
+    if @arguments.key?(:ids)
+      BadArgumentError
+    elsif return_all_submissions? && !partner?
+      AllSubmissionsError
+    elsif user_mismatch?
+      UserMismatchError
+    end
+  end
+
   def base_submissions
     @arguments[:available] ? Submission.available : Submission.all
   end
 
   def conditions
-    { id: @arguments[:ids], user_id: @arguments[:user_id] }.compact
+    { id: submission_ids.presence, user_id: user_ids.presence }.compact
+  end
+
+  def submission_ids
+    @arguments.fetch(:ids, [])
```

Comment:
What does it mean to execute `fetch` on `arguments`? Is this for providing defaults when a value is not found?

Reply:
Yep, that's exactly right! https://ruby-doc.org/core-2.5.1/Hash.html#method-i-fetch

These are (mostly) equal:

```ruby
@arguments.fetch(:ids, [])
@arguments[:ids] || []
```

Worth noting that fetch will return nils!

```ruby
irb(main):012:0> hsh = { omg: nil }
irb(main):013:0> hsh[:omg] || "lol"
=> "lol"
irb(main):014:0> hsh.fetch(:omg, "lol")
=> nil
```

So it sorta depends what you're trying to do, but I tend to reach for fetch first as I like how it reads better.